operators/multiclusterobservability/controllers/placementrule/manifestwork.go

```diff
@@ -568,9 +572,32 @@ func createUpdateResourcesForHubMetricsCollection(c client.Client, manifests []w
 					needsUpdate = true
 				}
 			case *corev1.ServiceAccount:
-				currentServiceAccount := currentObj.(*corev1.ServiceAccount)
-				if !reflect.DeepEqual(obj.ImagePullSecrets, currentServiceAccount.ImagePullSecrets) {
-					needsUpdate = true
+				// https://issues.redhat.com/browse/ACM-10967
+				// Some of these ServiceAccounts will be read from static files so they will never contain
+				// the generated Secrets as part of their corev1.ServiceAccount.ImagePullSecrets field.
+				// This checks by way of slice length if this particular ServiceAccount can be one of those.
```

Comment:
SHould we special case it for endpoint-sa? just to make sure no other SA runs through this logic, even if there isnt any other one

Reply:
I don't think so since we would see this problematic behaviour with other SAs too